sdk/enoki/src/react.tsx

```diff
@@ -67,3 +74,45 @@ export function useAuthCallback() {
 
 	return { handled, state };
 }
+
+export const EnokiWalletContext = createContext<{
+	wallets: ReturnType<typeof registerEnokiWallets>['wallets'];
+	client: SuiClient;
+} | null>(null);
+
+export function EnokiWalletProvider({
+	children,
+	config,
+	useSuiClientContext,
+}: {
+	config: Omit<Parameters<typeof registerEnokiWallets>[0], 'client'>;
+	useSuiClientContext: () => { client: SuiClient; network: string };
+	children: React.ReactNode;
+}) {
+	const { client, network } = useSuiClientContext();
+
+	const { wallets, unregister } = useMemo(
+		() => registerEnokiWallets({ ...config, client, network }),
+		[client, config, network],
+	);
+
+	useEffect(() => {
+		return () => {
+			unregister();
+		};
+	}, [unregister]);
+
+	return (
```

Comment:
you can memo-ize the context values as a slight performance optimization to guard against parent components re-rendering and causing a new object reference to be created + passed down through the context every time:

 ```
const values = useMemo(() => ({ wallets, client }));
```

(little un-intuitive so https://stackoverflow.com/a/62231233 for more info 😄)